kit/preprocess.js

```diff
@@ -117,6 +117,45 @@ export const docstringPreprocess = {
 	}
 };
 
+// Preprocessor that converts markdown into FrameworkContent
+// svelte component using mdsvexPreprocess
+export const frameorkcontentPreprocess = {
```

Comment:
using preprocessor to keep the syntax simpler
```html
<frameworkcontent>
<pytorch>
PYTORCH SPECIFIC CONTENT
</pytorch>
<tensorflow>
TENSORFLOW SPECIFIC CONTENT
</tensorflow>
<jax>
FLAX SPECIFIC CONTENT
</jax>
</frameworkcontent>
```

rather than letting a user write svelte with bunch of slots

Reply:
Maybe you could do that without a custom parser but with pure svelte code.

For example the `jax` component could be something like:

```ts
<FrameworkContentBlock framework="jax">
  <Markdown>
  	<slot />
  </Markdown>
</FrameworkContentBlock>
```

no?

Reply:
indeed, it would be possible to directly write svelte. There are 2 reasons why I prefer preprocessing:
1. `<frameworkcontent><jax>...</jax></frameworkcontent>` is simpler
2. Moreover, svelte/mdsvex compiler requires to have newlines so that a user can still write markdown inside slots. Thus `\n\n${fwContent}\n\n`, which is easy to do on preprocessor & hard to require if a user writes svelte directly
https://github.com/huggingface/doc-builder/blob/f3bb70cff4317f1feae4d0e134b5152cef7404c9/kit/preprocess.js#L143-L145

Reply:
What about this:

```html
<FrameworkContentBlock framework="jax">
  <Markdown>

<slot />

  </Markdown>
</FrameworkContentBlock>
```

But fair enough. Using svelte components would also require the user to order correctly the blocks, or add context logic with css ordering. And there is already existing code that preprocesses doc strings...